bin/transform

```diff
@@ -245,6 +253,60 @@ def update_metadata(curated_gisaid_data: pd.DataFrame) -> pd.DataFrame:
     return curated_gisaid_data
 
 
+def warn_future_dates(gisaid_data: pd.DataFrame):
+    """
+    Sends a Slack message detailing any GISAID EPI ISL IDs in a given
+    *gisaid_data* DataFrame that have collection dates from the future (e.g. a
+    date > now).
+    """
+    def future_date(row: pd.Series):
+        """
+        Inserts a key-value pair in the within-scope *future_dates* dictionary
+        for every GISAID EPI ISL ID and future date, respectively.
+        """
+        for datetime_format in ('%Y-%m-%d', '%Y-%m', '%Y'):
+            try:
+                date = datetime.strptime(row.date, datetime_format)
+            except ValueError:
+                continue
+
+            if date > datetime.now():
+                future_dates[row.gisaid_epi_isl] = row.date
```

Comment:
I think this whole bit could make use of the observation that ISO-8601 dates (even truncated ones) are string-comparable and skip the whole `strptime` step by directly comparing the current date to `row.date`. For example, consider the following:

```
In [1]: from datetime import date

In [2]: today = str(date.today())

In [3]: today
Out[3]: '2020-05-04'

In [4]: '2020-08' > today
Out[4]: True

In [5]: '2020-05' > today
Out[5]: False

In [6]: '2020-05-30' > today
Out[6]: True

In [7]: '2020-04' > today
Out[7]: False

In [8]: '2020' > today
Out[8]: False
```

NAs might have to be dropped first in the actual code.

Reply:
That's really clever! I'll incorporate that in a new commit.